src/NuGet.Core/NuGet.Commands/Utility/MSBuildProjectFrameworkUtility.cs

```diff
@@ -113,55 +198,47 @@ public static class MSBuildProjectFrameworkUtility
                     platformIdentifier = FrameworkConstants.FrameworkIdentifiers.Windows;
                 }
 
-                frameworks.Add($"{platformIdentifier}, Version={platformVersion}");
-
-                return frameworks;
+                return valueFactory($"{platformIdentifier}, Version={platformVersion}");
             }
 
             if (!string.IsNullOrEmpty(platformVersion)
                 && StringComparer.OrdinalIgnoreCase.Equals(platformIdentifier, "UAP"))
             {
                 // Use the platform id and versions, this is done for UAP projects
-                frameworks.Add($"{platformIdentifier}, Version={platformVersion}");
-
-                return frameworks;
+                return valueFactory($"{platformIdentifier}, Version={platformVersion}");
             }
 
             // TargetFrameworkMoniker
-            currentFrameworkString = MSBuildStringUtility.TrimAndGetNullForEmpty(targetFrameworkMoniker);
+            var currentFrameworkString = MSBuildStringUtility.TrimAndGetNullForEmpty(targetFrameworkMoniker);
+            var trimmedTargetFrameworkIdentifier = MSBuildStringUtility.TrimAndGetNullForEmpty(targetFrameworkIdentifier);
+            var trimmedTargetFrameworkVersion = MSBuildStringUtility.TrimAndGetNullForEmpty(targetFrameworkVersion);
+            var hasTFIandTFV = !string.IsNullOrEmpty(trimmedTargetFrameworkIdentifier) && !string.IsNullOrEmpty(trimmedTargetFrameworkVersion);
```

Comment:
very minor perf improvement possibility: since `MSBuildStringUtility.TrimAndGetNullForEmpty` sounds like it returns null when the input is an empty string or whitespace, you can change `string.IsNullOrEmpty` to normal null checks.

Reply:
fixed.